test/test.js

```diff
@@ -0,0 +1,124 @@
+        it('Can get the current price', async function () {
+            const config = getConfig();
+            let app = await buildApp(config);
+            return app.getPrice(config.authorization.apiKey).then((new_price) => {
+                console.log('<<<<<get current price>>>>', new_price);
+                assert.notEqual(new_price, null);
+            }).catch((error) => {
+                assert.fail(error);
+            })
+        })
+    });
+    describe('#run()', () => {
+        // Check if the keyboard key is the right color when the new price > the old price
+        it('Can store the last price and get it from storage', function () {
+            // This price has to be adapted
+            let app = new FlightPriceWatcher();
+            const price = '100';
+            console.log('<<<<<set new price>>>>', price);
+            assert.ok(app.setLastPrice(price));
+            assert.equal(app.getLastPrice(), price);
+        })
+        it('check the color of the key', async function () {
+            const config = getConfig();
+            let app = await buildApp(config);
+            return app.run().then((signal) => {
+                assert.ok(signal); 
+                // Has to be red
+                assert.equal(signal.points[0][0].color, '#DF0101');
+            }).catch((error) => {
+                assert.fail(error)
+            });
+        });
+    });
+    describe('#run()', () => {
+        // Check if the keyboard key is the right color when the new price <= the old price
+        it('Can store the last price and get it from storage', function () {
+            // This price has to be adapted
+            let app = new FlightPriceWatcher();
+            const price = '500';
+            console.log('<<<<<set new price>>>>', price);
+            assert.ok(app.setLastPrice(price));
+            assert.equal(app.getLastPrice(), price);
+        })
+        it('check the color of the key', async function () {
+            const config = getConfig();
+            let app = await buildApp(config);
+            return app.run().then((signal) => {
+                assert.ok(signal); 
+                // Has to be green
+                assert.equal(signal.points[0][0].color, '#088A08');
+            }).catch((error) => {
+                assert.fail(error)
+            });
+        });
+    });
+});
+
+function getConfig() {
+    const defaultConfig = {
+        applet: {
+            user: {
+                originPlace: 'JFK',
+                destinationPlace: 'AUS',
+                departDate: '2020-01-01',
+                returnDate: '2020-01-10',
+                currency: 'USD'
+            }            
+        },
+        geometry: {
+            width: 1,
+            height: 1,
+        },
+        authorization: {
+            // We can't hardcode an API KEY for safety reasons
+            apiKey: '' // 'ENTER THE API KEY IN ORDER TO TEST'  
```

Comment:
To test you can read the apiKey from a .json file that you never commit ( added to .gitignore) example test/credentials.json